File: 2026-01/src/day_31/boj_15649/Main.java
package day_31.boj_15649;

import java.io.*;
import java.util.*;

public class Main {

    static StringBuilder sb = new StringBuilder();
    static int n, m;
    static int[] sel;
    static boolean[] vis;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        sel = new int[m];
        vis = new boolean[1 + n];

        dfs(0);
        System.out.println(sb);
    }

    static void dfs(int sidx) {
        if (sidx == m) {
            for (int x : sel) {
                sb.append(x).append(" ");
            }
            sb.append("\n");
            return;
        }

        for (int i = 1; i <= n; i++) {
            if (vis[i]) continue;

            sel[sidx] = i;
            vis[i] = true;
            dfs(sidx + 1);
            vis[i] = false;
        }
    }
}

File: 2026-01/src/day_31/boj_15649/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int n, m;
int sel[8];
bool vis[9];

void dfs(int sidx) {
    if (sidx == m) {
        for (int i = 0; i < m; i++) {
            cout << sel[i] << ' ';
        }
        cout << '\n';
        return;
    }

    for (int i = 1; i <= n; i++) {
        if (vis[i]) continue;

        sel[sidx] = i;
        vis[i] = true;
        dfs(sidx + 1);
        vis[i] = false;
    }
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin >> n >> m;

    dfs(0);
}
